### PR TITLE
only publish docker image if version is not yet published

### PR DESCRIPTION
**Pull request by nichlaes**

### CODE REVIEW

**Comment by nichlaes** [comment]
@prasadtalasila this should fix the #854

**Comment by prasadtalasila** [comment]
@nichlaes can you make a minor correction in `client` or `servers/lib` directory and see if this is enforced? Or is there an easy way to check the functioning of this PR?

**Comment by prasadtalasila** [comment]
The `client/config/dev.js` requires correction. The updated content of this file is:

```js
if (typeof window !== 'undefined') {
  window.env = {
    REACT_APP_ENVIRONMENT: 'dev',
    REACT_APP_URL: 'http://localhost:4000/',
    REACT_APP_URL_BASENAME: '',
    REACT_APP_URL_DTLINK: '/lab',
    REACT_APP_URL_LIBLINK: '',
    REACT_APP_WORKBENCHLINK_VNCDESKTOP: '/tools/vnc/?password=vncpassword',
    REACT_APP_WORKBENCHLINK_VSCODE: '/tools/vscode/',
    REACT_APP_WORKBENCHLINK_JUPYTERLAB: '/lab',
    REACT_APP_WORKBENCHLINK_JUPYTERNOTEBOOK: '',

    REACT_APP_CLIENT_ID: '1be55736756190b3ace4c2c4fb19bde386d1dcc748d20b47ea8cfb5935b8446c',
    REACT_APP_AUTH_AUTHORITY: 'https://gitlab.com/',
    REACT_APP_REDIRECT_URI: 'http://localhost:4000/Library',
    REACT_APP_LOGOUT_REDIRECT_URI: 'http://localhost:4000/',
    REACT_APP_GITLAB_SCOPES: 'openid profile read_user read_repository api',
  };
};
```

**Comment by prasadtalasila** [comment]
The `servers/lib/DEVELOPER.md` requires correction in the following table.

```markdown
| OS      | Absolute Path                 | Relative Path |
| ------- | ----------------------------- | ------------- |
| Windows | C:\DTaaS\files                | ..\..\files  |
| Linux   | /Users/\<Username\>/DTaaS/files | ../../files   |
```

File: servers/lib/DEVELOPER.md
Comment (prasadtalasila) [inline servers/lib/DEVELOPER.md:1]: This problem has been resolved in PR #927. Please restore the file and rebase the branch.

File: client/config/dev.js
Comment (prasadtalasila) [inline client/config/dev.js:1]: This problem has been resolved in PR #927. Please restore the file and rebase the branch.

**Review by prasadtalasila — COMMENTED** [review]
@nichlaes Please also check issue #854 for more recent explanation of the problem.

**Comment by nichlaes** [comment]
@prasadtalasila I have reset the changes and tested it locally. Should we merge this and test it on the client?

**Pull request merged**